test/app/model/ChessEngine.spec.ts

```diff
@@ -142,4 +143,184 @@ describe('Chess Engine', () => {
       ),
     ).toThrowError('There is no piece on this square.');
   });
+
+  describe('Return player moves without those that cause his king to check', () => {
+    const whiteKing = new King(Side.WHITE);
+    const whiteRook = new Rook(Side.WHITE);
+    const blackKing = new King(Side.BLACK);
+    const blackRook = new Rook(Side.BLACK);
+    // const playerWhite = new Player(Side.WHITE);
+
+    it(`Should return the same possible moves array if the king's move doesn't cause his check.`, () => {
+      const boardWithPieces: SquareWithPiece = {
+        A2: whiteKing,
+        A8: blackKing,
+      };
+      const chessboard = new Chessboard(boardWithPieces);
+      const engine = new ChessEngine(chessboard);
+      const movedPiecePosition: Square = { column: 'A', row: 2 };
+      const possibleMovesBeforeFiltration = [
+        { column: 'A', row: 1 },
+        { column: 'A', row: 3 },
+        { column: 'B', row: 1 },
+        { column: 'B', row: 2 },
+        { column: 'B', row: 3 },
+      ];
+
+      const returnedResult = engine.returnPlayerMovesWithoutThoseThatCauseHisKingToCheck(movedPiecePosition);
```

Comment:
Aaa nie zauwazylem tego wczesniej!!!, bo teraz Ci sie wywalaja testy i zwrocilem uwage. Ja bym tego nie testowal. Bo to metoda prywatna :) Tylko przetestowac possibleMoves czy dobrze zwraca - bo tego tez uzywamy jako ChessModel.